src/argparse/argparse.lua

```diff
@@ -0,0 +1,35 @@
+local logger = require("logger/logger")
+
+local argparse = {}
+
+function argparse:parse(raw)
+  local args = {}
+
+  function parse_flag(flag)
+    for i=1,#raw do
+      if raw[i] == '--' .. flag then
+        args[flag] = true
+        return
+      end
+    end
+  end
+
+  function parse_value(option)
+    for i=1,#raw do
+      if raw[i] == '--' .. option then
+        if not raw[i + 1] then
```

Comment:
So arguments with values are sequentially stored in the arguments array. Does that mean we have to go to `#raw-1` above to avoid running off the end?

Reply:
Sort of, it's a table with a sequence of numbers for keys.

We don't have to worry about poking at raw[#raw + 1] because accessing a key that doesn't exist will return nil.  The loop should still inspect the last element because it might be an option that we're looking for, and if it is, we should let the user know they forgot the value.